seed/models/building_file.py

```diff
@@ -77,154 +77,161 @@ def str_to_file_type(cls, file_type):
+    def process(self, organization_id, cycle, property_view=None):
         """
         Process the building file that was uploaded and create the correct models for the object
 
         :param organization_id: integer, ID of organization
         :param cycle: object, instance of cycle object
+        :param property_view: PropertyView, if a property view already exists, this is it; if not, one will be created
         :return: list, [status, (PropertyView|None), messages]
         """
 
-        if self.file_type == self.BUILDINGSYNC:
-            bs = BuildingSync()
-            bs.import_file(self.file.path)
-            data, errors, messages = bs.process(BuildingSync.BRICR_STRUCT)
-
-            if data and not errors:
-                # subselect the data that are needed to create the PropertyState object
-                md = MappingData()
-                create_data = {"organization_id": organization_id}
-                extra_data = {}
-                for k, v in data.items():
-                    # Skip the keys that are for measures and reports and process later
-                    if k in ['measures', 'reports', 'scenarios']:
-                        continue
-
-                    if md.find_column('PropertyState', k):
-                        create_data[k] = v
-                    else:
-                        # TODO: break out columns in the extra data that should be part of the
-                        # PropertyState and which ones should be added to some other class that
-                        # doesn't exist yet.
-                        extra_data[k] = v
-                        # create columns, if needed, for the extra_data fields
-
-                        Column.objects.get_or_create(
-                            organization_id=organization_id,
-                            column_name=k,
-                            table_name='PropertyState',
-                            is_extra_data=True,
-                        )
-
-                # create a new propertystate for the objects
-                property_state = PropertyState.objects.create(**create_data)
-                property_state.extra_data = extra_data
-                property_state.save()
-
-                # automatically promote this buildingsync file to a new instance
-                property_view = property_state.promote(cycle)
-
-                # set the property_state_id so that we can list the building files by properties
-                self.property_state_id = property_state.id
-                self.save()
-
-                # add in the measures
-                for m in data['measures']:
-                    # Find the measure in the database
-                    measure = Measure.objects.get(
-                        category=m['category'], name=m['name'], organization_id=organization_id,
-                    )
+        if self.file_type != self.BUILDINGSYNC:
+            return False, None, "File format was not set to BuildingSync"
```

Comment:
Gross, this diff looks worse than it is.  I tried to make it easier to chop up the main pieces of the function by replacing the if / else with some early return calls.  The only real change is [here](https://github.com/SEED-platform/seed/pull/1468/files#diff-74afe00e18809e72419ed1db0d6d2009R125) where I check if property_view is passed in, and if so I use that for further assignments.  If it's not, we go down the original path of creating a new state and view.

Reply:
Eek. yeah, I was about to flip out.